braintree/util/http.py

```diff
@@ -120,10 +122,14 @@ def http_do(self, http_verb, path, headers, request_body):
         return [response.status_code, response.text]
 
     def handle_exception(self, exception):
-        if isinstance(exception, requests.exceptions.ConnectionError):
+        if isinstance(exception, requests.exceptions.ReadTimeout):
+            raise ReadTimeoutError(exception)
+        elif isinstance(exception, requests.exceptions.ConnectionError):
             raise ConnectionError(exception)
         elif isinstance(exception, requests.exceptions.HTTPError):
             raise InvalidResponseError(exception)
+        elif isinstance(exception, requests.exceptions.ConnectTimeout):
```

Comment:
I don't think it can ever get here. I think this is because a ConnectTimeout is a child class of ConnectionError.

The unit tests I included in my other comment illustrate that it is raising a ConnectionError instead of the ConnectionTimeout error.

Reply:
Yeah you are right. I opted to move this above `ConnectionTimeout`. This branch was pretty old but that's what we ended up doing locally.